# Anime Dataset Visualisation




![](https://assets-prd.ignimgs.com/2022/08/17/top25animecharacters-blogroll-1660777571580.jpg)

In [1]:
import pandas as pd


In [2]:
url = "https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2019/2019-04-23/raw_anime.csv"
df = pd.read_csv(url)


In [3]:
df.head()

animeID                             name            title_english  \
0        1                     Cowboy Bebop             Cowboy Bebop   
1        5  Cowboy Bebop: Tengoku no Tobira  Cowboy Bebop: The Movie   
2        6                           Trigun                   Trigun   
3        7               Witch Hunter Robin       Witch Hunter Robin   
4        8                   Bouken Ou Beet   Beet the Vandel Buster   

       title_japanese                               title_synonyms   type  \
0           カウボーイビバップ                                           []     TV   
1      カウボーイビバップ 天国の扉  ["Cowboy Bebop: Knockin' on Heaven's Door"]  Movie   
2               トライガン                                           []     TV   
3  Witch Hunter ROBIN                                      ['WHR']     TV   
4              冒険王ビィト                      ['Adventure King Beet']     TV   

     source                     producers  \
0  Original             ['Bandai Visual']   
1  Original  ['Sunrise', 'Bandai Visual']   
2     Manga      ['Victor Entertainment']   
3  Original             ['Bandai Visual']   
4     Manga        ['TV Tokyo', 'Dentsu']   

                                               genre              studio  ...  \
0  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...         ['Sunrise']  ...   
1  ['Action', 'Drama', 'Mystery', 'Sci-Fi', 'Space']           ['Bones']  ...   
2  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...        ['Madhouse']  ...   
3  ['Action', 'Magic', 'Police', 'Supernatural', ...         ['Sunrise']  ...   
4  ['Adventure', 'Fantasy', 'Shounen', 'Supernatu...  ['Toei Animation']  ...   

   scored_by    rank popularity   members favorites  \
0   405664.0    26.0       39.0  795733.0   43460.0   
1   120243.0   164.0      449.0  197791.0     776.0   
2   212537.0   255.0      146.0  408548.0   10432.0   
3    32837.0  2371.0     1171.0   79397.0     537.0   
4     4894.0  3544.0     3704.0   11708.0      14.0   

                                            synopsis  \
0  In the year 2071, humanity has colonized sever...   
1  Another day, another bounty—such is the life o...   
2  Vash the Stampede is the man with a $$60,000,0...   
3  Witches are individuals with special powers li...   
4  It is the dark century and the people are suff...   

                                          background    premiered  \
0  When Cowboy Bebop first aired in spring of 199...  Spring 1998   
1                                                NaN          NaN   
2  The Japanese release by Victor Entertainment h...  Spring 1998   
3                                                NaN  Summer 2002   
4                                                NaN    Fall 2004   

                  broadcast                                            related  
0  Saturdays at 01:00 (JST)  {'Adaptation': [{'mal_id': 173, 'type': 'manga...  
1                       NaN  {'Parent story': [{'mal_id': 1, 'type': 'anime...  
2  Thursdays at 01:15 (JST)  {'Adaptation': [{'mal_id': 703, 'type': 'manga...  
3       Tuesdays at Unknown                                                 {}  
4  Thursdays at 18:30 (JST)  {'Adaptation': [{'mal_id': 1348, 'type': 'mang...  

[5 rows x 27 columns]

In [4]:
df.columns

Index(['animeID', 'name', 'title_english', 'title_japanese', 'title_synonyms',
       'type', 'source', 'producers', 'genre', 'studio', 'episodes', 'status',
       'airing', 'aired', 'duration', 'rating', 'score', 'scored_by', 'rank',
       'popularity', 'members', 'favorites', 'synopsis', 'background',
       'premiered', 'broadcast', 'related'],
      dtype='object')

In [5]:
df.dtypes

animeID             int64
name               object
title_english      object
title_japanese     object
title_synonyms     object
type               object
source             object
producers          object
genre              object
studio             object
episodes          float64
status             object
airing             object
aired              object
duration           object
rating             object
score             float64
scored_by         float64
rank              float64
popularity        float64
members           float64
favorites         float64
synopsis           object
background         object
premiered          object
broadcast          object
related            object
dtype: object

### Checking for null Values

In [6]:
df.shape[0]

15278

In [7]:
df.isnull().sum()

animeID               0
name                  0
title_english      9156
title_japanese       48
title_synonyms        5
type                  5
source                5
producers             5
genre                 5
studio                5
episodes            546
status                5
airing                5
aired                 5
duration              5
rating                5
score               500
scored_by             5
rank               1609
popularity            5
members               5
favorites             5
synopsis            713
background        14160
premiered         11099
broadcast         10876
related               5
dtype: int64

### We can see `title_synonyms`, `type`, `source`, `producers`, `genre`, `studio`, `status`, `airing`, `aired`, `duration`, `rating`, `scored_by`, `popularity`, `members`, `favorites`, `related` have 5 rows with Null values let drop rows from these column

In [8]:
# drop rows with missing values only in specific columns
df.dropna(subset=['title_synonyms', 'type', 'source', 'producers', 'genre', 'studio', 'status', 'airing', 'aired', 'duration', 'rating', 'scored_by', 'popularity', 'members', 'favorites', 'related'], inplace=True)

As we know we have `15278` rows and almost `14160` rows in `background` has Null value let's drop this column

In [9]:
df.drop('background', axis=1, inplace=True)

In [10]:
df.columns

Index(['animeID', 'name', 'title_english', 'title_japanese', 'title_synonyms',
       'type', 'source', 'producers', 'genre', 'studio', 'episodes', 'status',
       'airing', 'aired', 'duration', 'rating', 'score', 'scored_by', 'rank',
       'popularity', 'members', 'favorites', 'synopsis', 'premiered',
       'broadcast', 'related'],
      dtype='object')

In [11]:
df.isnull().sum()

animeID               0
name                  0
title_english      9151
title_japanese       43
title_synonyms        0
type                  0
source                0
producers             0
genre                 0
studio                0
episodes            541
status                0
airing                0
aired                 0
duration              0
rating                0
score               495
scored_by             0
rank               1604
popularity            0
members               0
favorites             0
synopsis            708
premiered         11094
broadcast         10871
related               0
dtype: int64

We can see Main columns have lots of missing values as such:
- `title_english` has almost `9151` missing value but we need this names for visualisation so let's try to translate name of `title_japanese` to english for these values

In [12]:
!pip install googletrans==4.0.0-rc1


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [13]:
from googletrans import Translator
translator = Translator()

for i, row in df.iterrows():
    if pd.isna(row['title_english']):
        try:
            # translate 'title_japanese' to english
            translated = translator.translate(row['title_japanese'], dest='en').text
            # fill missing values in 'title_english' column
            df.at[i, 'title_english'] = translated
        except:
            # handle any errors
            print(f"Translation failed for row {i}")


Translation failed for row 1422
Translation failed for row 1435
Translation failed for row 1439
Translation failed for row 1442
Translation failed for row 1454
Translation failed for row 1456
Translation failed for row 1463
Translation failed for row 1470
Translation failed for row 1476
Translation failed for row 1484
Translation failed for row 1493
Translation failed for row 1499
Translation failed for row 1501
Translation failed for row 1503
Translation failed for row 1508
Translation failed for row 1525
Translation failed for row 1532
Translation failed for row 1535
Translation failed for row 1549
Translation failed for row 1550
Translation failed for row 1553
Translation failed for row 1556
Translation failed for row 1558
Translation failed for row 1559
Translation failed for row 1562
Translation failed for row 1569
Translation failed for row 1575
Translation failed for row 1578
Translation failed for row 1580
Translation failed for row 1581
Translation failed for row 1582
Translat

In [14]:
df.isnull().sum()

animeID               0
name                  0
title_english      4214
title_japanese       43
title_synonyms        0
type                  0
source                0
producers             0
genre                 0
studio                0
episodes            541
status                0
airing                0
aired                 0
duration              0
rating                0
score               495
scored_by             0
rank               1604
popularity            0
members               0
favorites             0
synopsis            708
premiered         11094
broadcast         10871
related               0
dtype: int64

## Saving Data

In [15]:
df.to_csv('preprocessed.csv')

### Loading Translated Dataset

In [47]:
path = "/content/preprocessed.csv"
translated_anime = pd.read_csv(path)

translated_anime.head(3)

Unnamed: 0  animeID                             name  \
0           0        1                     Cowboy Bebop   
1           1        5  Cowboy Bebop: Tengoku no Tobira   
2           2        6                           Trigun   

             title_english  title_japanese  \
0             Cowboy Bebop       カウボーイビバップ   
1  Cowboy Bebop: The Movie  カウボーイビバップ 天国の扉   
2                   Trigun           トライガン   

                                title_synonyms   type    source  \
0                                           []     TV  Original   
1  ["Cowboy Bebop: Knockin' on Heaven's Door"]  Movie  Original   
2                                           []     TV     Manga   

                      producers  \
0             ['Bandai Visual']   
1  ['Sunrise', 'Bandai Visual']   
2      ['Victor Entertainment']   

                                               genre  ... score  scored_by  \
0  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...  ...  8.81   405664.0   
1  ['Action', 'Drama', 'Mystery', 'Sci-Fi', 'Space']  ...  8.41   120243.0   
2  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...  ...  8.30   212537.0   

    rank  popularity   members favorites  \
0   26.0        39.0  795733.0   43460.0   
1  164.0       449.0  197791.0     776.0   
2  255.0       146.0  408548.0   10432.0   

                                            synopsis    premiered  \
0  In the year 2071, humanity has colonized sever...  Spring 1998   
1  Another day, another bounty—such is the life o...          NaN   
2  Vash the Stampede is the man with a $$60,000,0...  Spring 1998   

                  broadcast                                            related  
0  Saturdays at 01:00 (JST)  {'Adaptation': [{'mal_id': 173, 'type': 'manga...  
1                       NaN  {'Parent story': [{'mal_id': 1, 'type': 'anime...  
2  Thursdays at 01:15 (JST)  {'Adaptation': [{'mal_id': 703, 'type': 'manga...  

[3 rows x 27 columns]

In [48]:
translated_anime.isnull().sum()

Unnamed: 0            0
animeID               0
name                  0
title_english      4214
title_japanese       43
title_synonyms        0
type                  0
source                0
producers             0
genre                 0
studio                0
episodes            541
status                0
airing                0
aired                 0
duration              0
rating                0
score               495
scored_by             0
rank               1604
popularity            0
members               0
favorites             0
synopsis            708
premiered         11094
broadcast         10871
related               0
dtype: int64

In [49]:
translated_anime.shape[0]

15273

In [50]:
# drop rows with missing values only in specific columns
translated_anime.dropna(subset=['title_english', 'title_japanese', 'episodes','score','rank','synopsis','premiered','broadcast'], inplace=True)

In [51]:
translated_anime.columns

Index(['Unnamed: 0', 'animeID', 'name', 'title_english', 'title_japanese',
       'title_synonyms', 'type', 'source', 'producers', 'genre', 'studio',
       'episodes', 'status', 'airing', 'aired', 'duration', 'rating', 'score',
       'scored_by', 'rank', 'popularity', 'members', 'favorites', 'synopsis',
       'premiered', 'broadcast', 'related'],
      dtype='object')

In [52]:
translated_anime.head(3)

Unnamed: 0  animeID                name       title_english  \
0           0        1        Cowboy Bebop        Cowboy Bebop   
2           2        6              Trigun              Trigun   
3           3        7  Witch Hunter Robin  Witch Hunter Robin   

       title_japanese title_synonyms type    source                 producers  \
0           カウボーイビバップ             []   TV  Original         ['Bandai Visual']   
2               トライガン             []   TV     Manga  ['Victor Entertainment']   
3  Witch Hunter ROBIN        ['WHR']   TV  Original         ['Bandai Visual']   

                                               genre  ... score  scored_by  \
0  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...  ...  8.81   405664.0   
2  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...  ...  8.30   212537.0   
3  ['Action', 'Magic', 'Police', 'Supernatural', ...  ...  7.33    32837.0   

     rank  popularity   members favorites  \
0    26.0        39.0  795733.0   43460.0   
2   255.0       146.0  408548.0   10432.0   
3  2371.0      1171.0   79397.0     537.0   

                                            synopsis    premiered  \
0  In the year 2071, humanity has colonized sever...  Spring 1998   
2  Vash the Stampede is the man with a $$60,000,0...  Spring 1998   
3  Witches are individuals with special powers li...  Summer 2002   

                  broadcast                                            related  
0  Saturdays at 01:00 (JST)  {'Adaptation': [{'mal_id': 173, 'type': 'manga...  
2  Thursdays at 01:15 (JST)  {'Adaptation': [{'mal_id': 703, 'type': 'manga...  
3       Tuesdays at Unknown                                                 {}  

[3 rows x 27 columns]

### Removing Redundant Columns
 - `Unnamed: 0`
 - `animeID`
 - `title_synonyms`
 - `members`
 - `synopsis` 
 - `related`

In [53]:
translated_anime.drop(['Unnamed: 0','animeID','title_synonyms','members','synopsis','related'],axis = 1, inplace = True)


In [54]:
translated_anime.head(3)

name       title_english      title_japanese type    source  \
0        Cowboy Bebop        Cowboy Bebop           カウボーイビバップ   TV  Original   
2              Trigun              Trigun               トライガン   TV     Manga   
3  Witch Hunter Robin  Witch Hunter Robin  Witch Hunter ROBIN   TV  Original   

                  producers  \
0         ['Bandai Visual']   
2  ['Victor Entertainment']   
3         ['Bandai Visual']   

                                               genre        studio  episodes  \
0  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...   ['Sunrise']      26.0   
2  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...  ['Madhouse']      26.0   
3  ['Action', 'Magic', 'Police', 'Supernatural', ...   ['Sunrise']      26.0   

            status  ...                                              aired  \
0  Finished Airing  ...  {'from': '1998-04-03T00:00:00+00:00', 'to': '1...   
2  Finished Airing  ...  {'from': '1998-04-01T00:00:00+00:00', 'to': '1...   
3  Finished Airing  ...  {'from': '2002-07-02T00:00:00+00:00', 'to': '2...   

        duration                          rating score  scored_by    rank  \
0  24 min per ep  R - 17+ (violence & profanity)  8.81   405664.0    26.0   
2  24 min per ep       PG-13 - Teens 13 or older  8.30   212537.0   255.0   
3  25 min per ep       PG-13 - Teens 13 or older  7.33    32837.0  2371.0   

   popularity  favorites    premiered                 broadcast  
0        39.0    43460.0  Spring 1998  Saturdays at 01:00 (JST)  
2       146.0    10432.0  Spring 1998  Thursdays at 01:15 (JST)  
3      1171.0      537.0  Summer 2002       Tuesdays at Unknown  

[3 rows x 21 columns]

In [55]:
translated_anime.columns

Index(['name', 'title_english', 'title_japanese', 'type', 'source',
       'producers', 'genre', 'studio', 'episodes', 'status', 'airing', 'aired',
       'duration', 'rating', 'score', 'scored_by', 'rank', 'popularity',
       'favorites', 'premiered', 'broadcast'],
      dtype='object')

In [56]:
translated_anime.drop(['status','aired','airing'], axis=1, inplace=True)

In [57]:
translated_anime.head(3)

name       title_english      title_japanese type    source  \
0        Cowboy Bebop        Cowboy Bebop           カウボーイビバップ   TV  Original   
2              Trigun              Trigun               トライガン   TV     Manga   
3  Witch Hunter Robin  Witch Hunter Robin  Witch Hunter ROBIN   TV  Original   

                  producers  \
0         ['Bandai Visual']   
2  ['Victor Entertainment']   
3         ['Bandai Visual']   

                                               genre        studio  episodes  \
0  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...   ['Sunrise']      26.0   
2  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...  ['Madhouse']      26.0   
3  ['Action', 'Magic', 'Police', 'Supernatural', ...   ['Sunrise']      26.0   

        duration                          rating  score  scored_by    rank  \
0  24 min per ep  R - 17+ (violence & profanity)   8.81   405664.0    26.0   
2  24 min per ep       PG-13 - Teens 13 or older   8.30   212537.0   255.0   
3  25 min per ep       PG-13 - Teens 13 or older   7.33    32837.0  2371.0   

   popularity  favorites    premiered                 broadcast  
0        39.0    43460.0  Spring 1998  Saturdays at 01:00 (JST)  
2       146.0    10432.0  Spring 1998  Thursdays at 01:15 (JST)  
3      1171.0      537.0  Summer 2002       Tuesdays at Unknown

In [58]:
translated_anime.columns

Index(['name', 'title_english', 'title_japanese', 'type', 'source',
       'producers', 'genre', 'studio', 'episodes', 'duration', 'rating',
       'score', 'scored_by', 'rank', 'popularity', 'favorites', 'premiered',
       'broadcast'],
      dtype='object')

In [59]:
translated_anime[['premiered_season', 'premiered_year']] = translated_anime['premiered'].str.split(' ', n=1, expand=True)
translated_anime.drop(['premiered'], axis=1, inplace=True)
translated_anime.head(3)

name       title_english      title_japanese type    source  \
0        Cowboy Bebop        Cowboy Bebop           カウボーイビバップ   TV  Original   
2              Trigun              Trigun               トライガン   TV     Manga   
3  Witch Hunter Robin  Witch Hunter Robin  Witch Hunter ROBIN   TV  Original   

                  producers  \
0         ['Bandai Visual']   
2  ['Victor Entertainment']   
3         ['Bandai Visual']   

                                               genre        studio  episodes  \
0  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...   ['Sunrise']      26.0   
2  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...  ['Madhouse']      26.0   
3  ['Action', 'Magic', 'Police', 'Supernatural', ...   ['Sunrise']      26.0   

        duration                          rating  score  scored_by    rank  \
0  24 min per ep  R - 17+ (violence & profanity)   8.81   405664.0    26.0   
2  24 min per ep       PG-13 - Teens 13 or older   8.30   212537.0   255.0   
3  25 min per ep       PG-13 - Teens 13 or older   7.33    32837.0  2371.0   

   popularity  favorites                 broadcast premiered_season  \
0        39.0    43460.0  Saturdays at 01:00 (JST)           Spring   
2       146.0    10432.0  Thursdays at 01:15 (JST)           Spring   
3      1171.0      537.0       Tuesdays at Unknown           Summer   

  premiered_year  
0           1998  
2           1998  
3           2002

In [60]:
translated_anime.dtypes

name                 object
title_english        object
title_japanese       object
type                 object
source               object
producers            object
genre                object
studio               object
episodes            float64
duration             object
rating               object
score               float64
scored_by           float64
rank                float64
popularity          float64
favorites           float64
broadcast            object
premiered_season     object
premiered_year       object
dtype: object

In [61]:
translated_anime.isnull().sum()

name                0
title_english       0
title_japanese      0
type                0
source              0
producers           0
genre               0
studio              0
episodes            0
duration            0
rating              0
score               0
scored_by           0
rank                0
popularity          0
favorites           0
broadcast           0
premiered_season    0
premiered_year      0
dtype: int64

In [62]:
translated_anime.shape[0]

3295

In [63]:
translated_anime['producers'].unique()

array(["['Bandai Visual']", "['Victor Entertainment']",
       "['TV Tokyo', 'Dentsu']", ...,
       "['Studio Mausu', 'Namu Animation']",
       "['DAX Production', 'Twin Planet']", "['Polygon Magic']"],
      dtype=object)

we first use the ast.literal_eval() function to convert each string representation of a list to an actual list of strings. Then, we use the .str accessor to get the first element of each list.

In [64]:
import ast

# Convert 'producers' column to list data type
translated_anime['producers'] = translated_anime['producers'].apply(lambda x: ast.literal_eval(x) if isinstance(x, str) else x)

# Extract the first element from each list in 'producers' column
translated_anime['producers'] = translated_anime['producers'].str[0]


In [65]:
translated_anime.head(3)

name       title_english      title_japanese type    source  \
0        Cowboy Bebop        Cowboy Bebop           カウボーイビバップ   TV  Original   
2              Trigun              Trigun               トライガン   TV     Manga   
3  Witch Hunter Robin  Witch Hunter Robin  Witch Hunter ROBIN   TV  Original   

              producers                                              genre  \
0         Bandai Visual  ['Action', 'Adventure', 'Comedy', 'Drama', 'Sc...   
2  Victor Entertainment  ['Action', 'Sci-Fi', 'Adventure', 'Comedy', 'D...   
3         Bandai Visual  ['Action', 'Magic', 'Police', 'Supernatural', ...   

         studio  episodes       duration                          rating  \
0   ['Sunrise']      26.0  24 min per ep  R - 17+ (violence & profanity)   
2  ['Madhouse']      26.0  24 min per ep       PG-13 - Teens 13 or older   
3   ['Sunrise']      26.0  25 min per ep       PG-13 - Teens 13 or older   

   score  scored_by    rank  popularity  favorites                 broadcast  \
0   8.81   405664.0    26.0        39.0    43460.0  Saturdays at 01:00 (JST)   
2   8.30   212537.0   255.0       146.0    10432.0  Thursdays at 01:15 (JST)   
3   7.33    32837.0  2371.0      1171.0      537.0       Tuesdays at Unknown   

  premiered_season premiered_year  
0           Spring           1998  
2           Spring           1998  
3           Summer           2002

we first use the ast.literal_eval() function to convert each string representation of a list to an actual list of strings. Then, we use the random.randint() function to generate a random integer between 0 and the length of the list of genres (minus one, since indexing starts at zero). Finally, we use this random integer to select a random genre from the list using indexing, and assign it to the 'genre' column.

Note that we use if x to check if the 'genre' column contains any empty lists, and if it does, we assign an empty string to that row's 'genre' value.

In [66]:
import random

# Convert 'genre' column to list data type
translated_anime['genre'] = translated_anime['genre'].apply(lambda x: ast.literal_eval(x) if isinstance(x, str) else x)

# Randomly select a genre from each list in 'genre' column
translated_anime['genre'] = translated_anime['genre'].apply(lambda x: x[random.randint(0, len(x)-1)] if x else '')
translated_anime.head(3)

name       title_english      title_japanese type    source  \
0        Cowboy Bebop        Cowboy Bebop           カウボーイビバップ   TV  Original   
2              Trigun              Trigun               トライガン   TV     Manga   
3  Witch Hunter Robin  Witch Hunter Robin  Witch Hunter ROBIN   TV  Original   

              producers   genre        studio  episodes       duration  \
0         Bandai Visual  Sci-Fi   ['Sunrise']      26.0  24 min per ep   
2  Victor Entertainment  Sci-Fi  ['Madhouse']      26.0  24 min per ep   
3         Bandai Visual   Magic   ['Sunrise']      26.0  25 min per ep   

                           rating  score  scored_by    rank  popularity  \
0  R - 17+ (violence & profanity)   8.81   405664.0    26.0        39.0   
2       PG-13 - Teens 13 or older   8.30   212537.0   255.0       146.0   
3       PG-13 - Teens 13 or older   7.33    32837.0  2371.0      1171.0   

   favorites                 broadcast premiered_season premiered_year  
0    43460.0  Saturdays at 01:00 (JST)           Spring           1998  
2    10432.0  Thursdays at 01:15 (JST)           Spring           1998  
3      537.0       Tuesdays at Unknown           Summer           2002

In [67]:
# Convert 'producers' column to list data type
translated_anime['studio'] = translated_anime['studio'].apply(lambda x: ast.literal_eval(x) if isinstance(x, str) else x)

# Extract the first element from each list in 'producers' column
translated_anime['studio'] = translated_anime['studio'].str[0]

In [68]:
translated_anime.head(3)

name       title_english      title_japanese type    source  \
0        Cowboy Bebop        Cowboy Bebop           カウボーイビバップ   TV  Original   
2              Trigun              Trigun               トライガン   TV     Manga   
3  Witch Hunter Robin  Witch Hunter Robin  Witch Hunter ROBIN   TV  Original   

              producers   genre    studio  episodes       duration  \
0         Bandai Visual  Sci-Fi   Sunrise      26.0  24 min per ep   
2  Victor Entertainment  Sci-Fi  Madhouse      26.0  24 min per ep   
3         Bandai Visual   Magic   Sunrise      26.0  25 min per ep   

                           rating  score  scored_by    rank  popularity  \
0  R - 17+ (violence & profanity)   8.81   405664.0    26.0        39.0   
2       PG-13 - Teens 13 or older   8.30   212537.0   255.0       146.0   
3       PG-13 - Teens 13 or older   7.33    32837.0  2371.0      1171.0   

   favorites                 broadcast premiered_season premiered_year  
0    43460.0  Saturdays at 01:00 (JST)           Spring           1998  
2    10432.0  Thursdays at 01:15 (JST)           Spring           1998  
3      537.0       Tuesdays at Unknown           Summer           2002

## Saving Dataset in CSV Format

In [70]:
translated_anime.to_csv('anime_dataset.csv')